frame/system/src/extensions/check_weight.rs

```diff
@@ -144,18 +145,18 @@ where
 
 	// Check if we don't exceed per-class allowance
 	match limit_per_class.max_total {
-		Some(max) if per_class > max => return Err(InvalidTransaction::ExhaustsResources.into()),
+		Some(max) if per_class.any_gt(max) => return Err(InvalidTransaction::ExhaustsResources.into()),
 		// There is no `max_total` limit (`None`),
 		// or we are below the limit.
 		_ => {},
 	}
 
 	// In cases total block weight is exceeded, we need to fall back
 	// to `reserved` pool if there is any.
-	if all_weight.total() > maximum_weight.max_block {
+	if all_weight.total().any_gt(maximum_weight.max_block) {
 		match limit_per_class.reserved {
 			// We are over the limit in reserved pool.
-			Some(reserved) if per_class > reserved =>
+			Some(reserved) if per_class.any_gt(reserved) =>
```

Comment:
This might be problematic, because some solo chains don't care about the storage size and only the computational time component... one solution that I have in mind is to add a feature flag for ignoring PoV sizes in weights, that then would allow the code above to still work as intended.

Reply:
Features should be our last resort. They are really annoying to deal with and not a good fit here (is standalone vs parachain really additive?). But since we decided `Weight` shouldn't be generic it is the only way. IMHO we should rethink that. Having `Weight` as an associated type on `frame_system::Config` wouldn't probably introduce too many new type parameters as anything is generic over `T: Config` anyways.

Reply:
Would be not disable size metering for solo chains at all? Aka all Weights on solo chains should have `size = 0`?

Otherwise, adding some feature doesn't sound like a good idea.

Reply:
Making the `size = 0`, or `max-size = infinity` should allow solo-chains.

👎 to features, as they have bitten us in the past.

Reply:
Okay, then the code above should be fine as-is, it's then up to the solo-chain maintainer to put an appropriate value for the max size allowed?